src/promql/src/error.rs

```diff
@@ -15,13 +15,42 @@
 use std::any::Any;
 
 use common_error::prelude::*;
+use promql_parser::parser::Expr as PromExpr;
 
 #[derive(Debug, Snafu)]
 #[snafu(visibility(pub))]
 pub enum Error {
     #[snafu(display("Unsupported expr type: {}", name))]
     UnsupportedExpr { name: String, backtrace: Backtrace },
 
+    #[snafu(display("DataFusion error: {}", source))]
+    DataFusion {
```

Comment:
Looks like using this error everywhere, but prefer to differentiate them by an error message. For example, when`get_table_provider` returns error, we can add a message `failed to find the table : {table_name}`. 

I think a detailed message would be better for users.